release notes/v0.4.0.md

```diff
@@ -0,0 +1,169 @@
+xk6-browser v0.4.0 is here! :tada:
+
+In this minor release, we implemented a large portion of a new API that will help you find elements on a page in a more robust way, fixed critical bugs, and made some improvements and code refactoring!
```

Comment:
WDYT? I think it should be prominent here at the start, otherwise it likely won't be read.

```suggestion
In this minor release, we implemented a large portion of a new API that will help you find elements on a page in a more robust way, fixed critical bugs, and made some improvements and code refactoring!

The team is aware that the extension is currently unstable for many users. This stems from the complexity in how [CDP](https://chromedevtools.github.io/devtools-protocol/) events from the browser and from our own internal event-based system are currently handled, as well as our lack of async support in JavaScript APIs, which introduces race conditions and can lead to intermittent failures. We've been delaying tackling these issues since it requires a substantial rearchitecture of core systems, but rest assured, it's currently our top priority and these issues will be addressed by, hopefully, the next release. If you want to follow along the progress, subscribe to issues [#427](https://github.com/grafana/xk6-browser/issues/427) and [#428](https://github.com/grafana/xk6-browser/issues/428).
```